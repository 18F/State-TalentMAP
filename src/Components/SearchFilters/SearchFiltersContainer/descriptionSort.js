const descriptionSort = (a, b) => {
  const A = a.item.description.toLowerCase();
  const B = b.item.description.toLowerCase();
  if (A < B) { // sort string ascending
    return -1;
  }
  if (A > B) { return 1; }
  return 0; // default return value (no sorting)
};

export default descriptionSort;
