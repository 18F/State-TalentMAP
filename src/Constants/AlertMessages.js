export const NO_LANGUAGES = 'None listed';
export const NO_POST = 'None listed';
export const NO_POST_DIFFERENTIAL = 'None listed';
export const NO_DANGER_PAY = 'None listed';
export const NO_COLA = 'None listed';
export const NO_TOUR_OF_DUTY = 'None listed';
