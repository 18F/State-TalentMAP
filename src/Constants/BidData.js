// prop names from the API
export const APPROVED_PROP = 'approved';
export const CLOSED_PROP = 'closed';
export const DRAFT_PROP = 'draft';
export const DECLINED_PROP = 'declined';
export const HAND_SHAKE_ACCEPTED_PROP = 'handshake_accepted';
export const HAND_SHAKE_OFFERED_PROP = 'handshake_offered';
export const HAND_SHAKE_DECLINED_PROP = 'handshake_declined';
export const PRE_PANEL_PROP = 'pre_panel';
export const IN_PANEL_PROP = 'in_panel';
export const SUBMITTED_PROP = 'submitted';

export const DRAFT_NUMBER = 1;
export const SUBMITTED_NUMBER = 2;
export const HAND_SHAKE_OFFERED_NUMBER = 3;
export const HAND_SHAKE_ACCEPTED_NUMBER = 4;
export const IN_PANEL_NUMBER = 5;
export const APPROVED_NUMBER = 6;

// We make all of them functions so that we don't have to guess whethe to call a function or not
export const DRAFT_TITLE = () => ['Draft'];
export const SUBMIT_BID_ACTION_TITLE = () => ['Submit Bid'];
export const SUBMIT_BID_COMPLETE_TITLE = (bid = {}) => ['Bid Submitted', bid.submission_date];
export const HAND_SHAKE_EVALUATE_TITLE = () => ['Evaluate Bid'];
export const HAND_SHAKE_EVALUATING_TITLE = () => ['Bid being evaluated'];
export const HAND_SHAKE_OFFERED_TITLE = (bid = {}) => ['Handshake Offered', bid.hand_shake_offered_date];
export const HAND_SHAKE_ACCEPTED_TITLE = (bid = {}) => ['Handshake Accepted', bid.hand_shake_accepted_date];
export const PANEL_TITLE = (bid = {}) => ['Panel',
  `${bid.scheduled_panel_date ? `Scheduled: ${bid.scheduled_panel_date}` : ''}`,
  `${bid.panel_date ? `Panel date: ${bid.panel_date}` : ''}`];
export const APPROVAL_TITLE = () => ['Approval'];
