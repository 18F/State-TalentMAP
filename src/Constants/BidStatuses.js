// store any bid statuses from the API here
// class_name is used to pass to parent container for conditional css class name
export const APPROVED = { property: 'approved', class_name: 'approved', text: 'Approved' };
export const CLOSED = { property: 'closed', class_name: 'closed', text: 'Closed' };
export const DRAFT = { property: 'draft', class_name: 'draft', text: 'Draft Bid' };
export const DECLINED = { property: 'declined', class_name: 'declined', text: 'Bid Unapproved' };
export const HAND_SHAKE_ACCEPTED = { property: 'handshake_accepted', class_name: 'hand-shake-accepted', text: 'Handshake Accepted' };
export const HAND_SHAKE_OFFERED = { property: 'handshake_offered', class_name: 'hand-shake-offered', text: 'Handshake Offered' };
export const IN_PANEL = { property: 'in_panel', class_name: 'in-panel', text: 'In Panel' };
export const SUBMITTED = { property: 'submitted', class_name: 'submitted', text: 'Bid Submitted' };

const statusArray = [
  APPROVED, CLOSED, DRAFT, DECLINED, HAND_SHAKE_ACCEPTED, HAND_SHAKE_OFFERED, IN_PANEL, SUBMITTED,
];

// find the correct object based on status, then return the class_name
export const getStatusClass = (statusName) => {
  const matchingStatus = statusArray.find(s => s.property === statusName);
  return matchingStatus ? matchingStatus.class_name : '';
};
