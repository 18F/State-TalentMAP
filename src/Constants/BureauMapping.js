export const BUREAU_MAPPING = [
  /* {
    id: 1,
    description: 'Regional Bureaus',
    codeGroup: ['110000', '120000', '130000', '140000', '146000', '150000', '160000'],
    isRegional: true,
  }, */
  {
    id: '2',
    description: 'Arms Control and International Security',
    codeGroup: ['014540', '010164', '014000'],
  },
  {
    id: '3',
    description: 'Civilian Security, Democracy, and Human Rights',
    codeGroup: ['013000', '033510', '018000', '031000', '010113', '011810'],
  },
  {
    id: '4',
    description: 'Democratic Republic of Congo',
    codeGroup: ['010108', '010183', '010106', '010145'],
  },
  {
    id: '5',
    description: 'Economic Growth, Energy, and the Environment',
    codeGroup: ['040700', '041600', '011701', '021250'],
  },
  {
    id: '6',
    description: 'Management',
    codeGroup: ['201000', '041900', '170100', '010623', '016000', '241000', '012900', '012200'],
  },
  {
    id: '7',
    description: 'Public Diplomacy and Public Affairs',
    codeGroup: ['100100'],
  },
  {
    id: '8',
    description: 'Special Representatives, Envoys, Advisors & Coordinators',
    codeGroup: ['011210', '011702', '010173', '010114', '010135', '010170', '010163', '010140', '010182', '010179', '010160', '010181'],
  },
];

export const getBureauCodeGroup = () => {
  let codeGroup = [];
  BUREAU_MAPPING.forEach((group) => {
    codeGroup = codeGroup.concat(group.codeGroup);
  });
  return codeGroup.toString();
};
