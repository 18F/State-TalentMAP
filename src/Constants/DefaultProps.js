export const ACCORDION_SELECTION = { main: '', sub: '' };

export const DEFAULT_HOME_PAGE_POSITIONS = {
  isPopular: [],
  isNew: [],
};

export default ACCORDION_SELECTION;
