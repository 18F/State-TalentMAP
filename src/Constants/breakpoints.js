// should match responsive breakpoints in _variables.scss
const breakpoints = {
  screenSmMin: 600,
  get screenXsMax() { return this.screenSmMin - 1; },
  screenMdMin: 951,
  get screenSmMax() { return this.screenMdMin - 1; },
  screenLgMin: 1200,
  get screenMdMax() { return this.screenLgMin - 1; },
};

export default breakpoints;
