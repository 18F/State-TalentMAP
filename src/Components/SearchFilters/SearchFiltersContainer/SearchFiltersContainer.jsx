import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MultiSelectFilterContainer from '../MultiSelectFilterContainer/MultiSelectFilterContainer';
import MultiSelectFilter from '../MultiSelectFilter/MultiSelectFilter';
import BooleanFilterContainer from '../BooleanFilterContainer/BooleanFilterContainer';
import LanguageFilter from '../LanguageFilter/LanguageFilter';
import { FILTER_ITEMS_ARRAY, ACCORDION_SELECTION_OBJECT } from '../../../Constants/PropTypes';
import { descriptionSort } from '../../../utilities';

class SearchFiltersContainer extends Component {

  constructor(props) {
    super(props);
    this.onSetAccordion = this.onSetAccordion.bind(this);
    this.onSetAccordionLanguage = this.onSetAccordionLanguage.bind(this);
  }

  onBooleanFilterClick(isChecked, code, selectionRef) {
    const object = Object.assign({});
    object[selectionRef] = isChecked ? code : '';
    this.props.queryParamUpdate(object);
  }

  onSetAccordion(a, b) {
    this.props.setAccordion({ main: a, sub: b });
  }

  onSetAccordionLanguage(a) {
    this.props.setAccordion({ main: 'Language', sub: a });
  }
  render() {
    // get our boolean filter names
    const sortedBooleanNames = ['Post Differential', 'Danger Pay', 'COLA', 'Domestic'];

    // store filters in Map
    const booleanFiltersMap = new Map();
    this.props.filters.forEach((searchFilter) => {
      if (searchFilter.item.bool) {
        booleanFiltersMap.set(searchFilter.item.title, searchFilter);
      }
    });

    // sort boolean filters by sortedBooleanNames
    // pull from Map
    const booleanFilters = [];
    sortedBooleanNames.forEach((b) => {
      const filter = booleanFiltersMap.get(b);
      if (filter) {
        booleanFilters.push(filter);
      }
    });

    // get our normal multi-select filters
    const multiSelectFilterNames = ['region', 'skill', 'grade', 'tod'];

    // create map
    const multiSelectFilterMap = new Map();

    // pull filters from props and add to Map
    this.props.filters.forEach((f) => {
      if (multiSelectFilterNames.indexOf(f.item.description) > -1) {
        // extra handling for skill
        if (f.item.description === 'skill') {
          f.data.sort(descriptionSort);
        }
        // add to Map
        multiSelectFilterMap.set(f.item.description, f);
      }
    });

    // get our language filter, which we'll render differently
    const languageFilters = this.props.filters.find(
      searchFilter =>
        (
          searchFilter.item.description === 'language'
        ),
    );

    // make sure we have an object to use in case there were no languages passed down
    const languageFilter = languageFilters || { item: {} };

    // languageFilters should only have one object, so we simply call languageFilters[0]
    const languageFilterObject =
      { content:
        (<LanguageFilter
          key={languageFilter.item.title}
          item={languageFilter}
          selectedAccordion={this.props.selectedAccordion}
          queryParamUpdate={(l) => {
            this.props.queryParamUpdate({ [languageFilter.item.selectionRef]: l });
          }}
          setAccordion={this.onSetAccordionLanguage}
        />),
        title: languageFilter.item.title,
        id: `accordion-${languageFilter.item.title}`,
        expanded: languageFilter.item.title === this.props.selectedAccordion.main,
      };

    // adding filters based on multiSelectFilterNames
    const sortedFilters = [];
    multiSelectFilterNames.forEach((n) => {
      const item = multiSelectFilterMap.get(n);
      if (item) {
        sortedFilters.push(
          { content:
            (<MultiSelectFilter
              key={item.item.title}
              item={item}
              queryParamToggle={this.props.queryParamToggle}
            />),
            title: item.item.title,
            id: `accordion-${item.item.title}`,
            expanded: item.item.title === this.props.selectedAccordion.main,
          },
        );
      }
    });
    // add language last
    sortedFilters.push(languageFilterObject);

    return (
      <div>
        <MultiSelectFilterContainer
          setAccordion={this.onSetAccordion}
          multiSelectFilterList={sortedFilters}
        />
        <div className="boolean-filter-container">
          <BooleanFilterContainer
            filters={booleanFilters}
            onBooleanFilterClick={(e, code, ref, iterator, value) => {
              booleanFilters[iterator].data[0].isSelected = !value;
              this.onBooleanFilterClick(e, code, ref);
            }
            }
          />
        </div>
      </div>
    );
  }
}

SearchFiltersContainer.propTypes = {
  filters: FILTER_ITEMS_ARRAY.isRequired,
  queryParamUpdate: PropTypes.func.isRequired,
  queryParamToggle: PropTypes.func.isRequired,
  selectedAccordion: ACCORDION_SELECTION_OBJECT.isRequired,
  setAccordion: PropTypes.func.isRequired,
};

export default SearchFiltersContainer;
