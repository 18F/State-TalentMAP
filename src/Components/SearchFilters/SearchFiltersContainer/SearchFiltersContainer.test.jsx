import { shallow } from 'enzyme';
import React from 'react';
import toJSON from 'enzyme-to-json';
import SearchFiltersContainer from './SearchFiltersContainer';
import { ACCORDION_SELECTION } from '../../../Constants/DefaultProps';

describe('SearchFiltersContainerComponent', () => {
  const items = [{
    item: { title: 'title', selectionRef: 'ref' },
    data: [{ isSelected: true }],
  },
  {
    item: { title: 'COLA', selectionRef: 'ref2', bool: true },
    data: [{ isSelected: false }],
  },
  {
    item: { title: 'region', description: 'region', selectionRef: 'ref3' },
    data: [{ isSelected: false, long_description: 'test', short_description: 'test' }],
  },
  ];

  const accordion = ACCORDION_SELECTION;

  it('can receive props', () => {
    const wrapper = shallow(
      <SearchFiltersContainer
        queryParamUpdate={() => {}}
        queryParamToggle={() => {}}
        selectedAccordion={accordion}
        setAccordion={() => {}}
        filters={items}
        fetchMissionAutocomplete={() => {}}
        missionSearchResults={[]}
        fetchPostAutocomplete={() => {}}
        postSearchResults={[]}
      />,
    );
    expect(wrapper.instance().props.filters[0].item.title).toBe(items[0].item.title);
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <SearchFiltersContainer
        queryParamUpdate={() => {}}
        queryParamToggle={() => {}}
        selectedAccordion={accordion}
        setAccordion={() => {}}
        filters={items}
        fetchMissionAutocomplete={() => {}}
        missionSearchResults={[]}
        fetchPostAutocomplete={() => {}}
        postSearchResults={[]}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('can call the onBooleanFilterClick function', () => {
    const toggleValue = { value: null };
    const wrapper = shallow(
      <SearchFiltersContainer
        queryParamUpdate={(e) => { toggleValue.value = e; }}
        queryParamToggle={() => {}}
        selectedAccordion={accordion}
        setAccordion={() => {}}
        filters={items}
        fetchMissionAutocomplete={() => {}}
        missionSearchResults={[]}
        fetchPostAutocomplete={() => {}}
        postSearchResults={[]}
      />,
    );

    // check the filter
    wrapper.instance().onBooleanFilterClick(true, '0', 'skill');
    expect(toggleValue.value.skill).toBe('0');

    // un-check the filter
    wrapper.instance().onBooleanFilterClick(false, '0', 'skill');
    expect(toggleValue.value.skill).toBe('');
  });

  it('can call the onSetAccordion function', () => {
    const toggleValue = { a: null, b: null };
    const wrapper = shallow(
      <SearchFiltersContainer
        queryParamUpdate={() => {}}
        queryParamToggle={() => {}}
        selectedAccordion={accordion}
        setAccordion={(a) => { toggleValue.a = a.main; toggleValue.b = a.sub; }}
        filters={items}
        fetchMissionAutocomplete={() => {}}
        missionSearchResults={[]}
        fetchPostAutocomplete={() => {}}
        postSearchResults={[]}
      />,
    );

    wrapper.instance().onSetAccordion(1, 2);
    expect(toggleValue.a).toBe(1);
    expect(toggleValue.b).toBe(2);
  });

  it('can call the onSetAccordionLanguage function', () => {
    const toggleValue = { a: null, b: null };
    const wrapper = shallow(
      <SearchFiltersContainer
        queryParamUpdate={() => {}}
        queryParamToggle={() => {}}
        selectedAccordion={accordion}
        setAccordion={(a) => { toggleValue.a = a.main; toggleValue.b = a.sub; }}
        filters={items}
        fetchMissionAutocomplete={() => {}}
        missionSearchResults={[]}
        fetchPostAutocomplete={() => {}}
        postSearchResults={[]}
      />,
    );

    wrapper.instance().onSetAccordionLanguage(1);
    expect(toggleValue.a).toBe('Language');
    expect(toggleValue.b).toBe(1);
  });

  it('contains Language', () => {
    const filters = [{
      item: { title: 'language', selectionRef: 'ref', description: 'language' },
    }];

    const wrapper = shallow(
      <SearchFiltersContainer
        queryParamUpdate={() => {}}
        queryParamToggle={() => {}}
        selectedAccordion={accordion}
        setAccordion={() => {}}
        filters={filters}
        fetchMissionAutocomplete={() => {}}
        missionSearchResults={[]}
        fetchPostAutocomplete={() => {}}
        postSearchResults={[]}
      />,
    );
    expect(wrapper.instance().props.filters[0].item.description).toBe('language');
  });

  it('contains Skills', () => {
    const filters = [{
      item: { title: 'skill', selectionRef: 'ref', description: 'skill' },
      data: [
        {
          id: 1,
          code: '001',
          description: 'SKILL 1',
        },
        {
          id: 2,
          code: '002',
          description: 'SKILL 2',
        },
      ],
    }];

    const wrapper = shallow(
      <SearchFiltersContainer
        queryParamUpdate={() => {}}
        queryParamToggle={() => {}}
        selectedAccordion={accordion}
        setAccordion={() => {}}
        filters={filters}
        fetchMissionAutocomplete={() => {}}
        missionSearchResults={[]}
        fetchPostAutocomplete={() => {}}
        postSearchResults={[]}
      />,
    );
    expect(wrapper.instance().props.filters[0].item.description).toBe('skill');
  });
});
