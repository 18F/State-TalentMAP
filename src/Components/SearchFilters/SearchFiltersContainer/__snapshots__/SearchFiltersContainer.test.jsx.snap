// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFiltersContainerComponent matches snapshot 1`] = `
<div>
  <MultiSelectFilterContainer
    multiSelectFilterList={
      Array [
        Object {
          "content": <LanguageFilter
            item={
                  Object {
                        "item": Object {},
                      }
            }
            queryParamUpdate={[Function]}
            selectedAccordion={
                  Object {
                        "main": "",
                        "sub": "",
                      }
            }
            setAccordion={[Function]}
      />,
          "expanded": false,
          "id": "accordion-undefined",
          "title": undefined,
        },
      ]
    }
    setAccordion={[Function]}
  />
  <div
    className="boolean-filter-container"
  >
    <BooleanFilterContainer
      filters={Array []}
      onBooleanFilterClick={[Function]}
    />
  </div>
</div>
`;
